lib/rubocop/cop/lint/shadowed_exception.rb

```diff
@@ -77,7 +77,21 @@ def contains_multiple_levels_of_exceptions?(group)
             return !(group.size == 2 && group.include?(NilClass))
           end
 
-          group.combination(2).any? { |a, b| a && b && a <=> b }
+          group.combination(2).any? do |a, b|
+            compare_exceptions(a, b)
+          end
+        end
+
+        def compare_exceptions(a, b)
+          if system_call_error?(a) && system_call_error?(b)
+            a.const_get(:Errno) != b.const_get(:Errno)
```

Comment:
I'm not sure this will work.

I have the following:

```ruby
begin
  something
rescue Errno::EAGAIN, Errno::EWOULDBLOCK, Errno::ECONNABORTED, Errno::EPROTO, Errno::EINTR
  handle_exception
end
```

Now it will report an offense, because i.e. `Errno::EAGAIN::Errno != Errno::ECONNABORTED`, or am I missing something?

Reply:
Thanks for the feedback and reproduce code! This was not enough condition 💦 
I think that this issue could be solved with the following commit.
5b67c78